Allow kdump load a new kernel

zpytela opened the pull request.

zpytela committed:
Allow kdump load a new kernel

The kexec_load(2) system call loads a new kernel that can be executed
later by reboot(2).
Example usage is the kdump service triggered on a system crash which
needs to load and run another kernel to save memory dump.

The lockdown integrity is required for kexec of unsigned images.
CAP_SYS_BOOT is checked on entering the kexec_load_check() syscall.

The kernel_kexec_load() selinux-policy interface was added.

zpytela commented:
PR updated to include confidentiality for reading /proc/kcore.

WOnder93 commented on policy/modules/kernel/kernel.if:
```diff
@@ -1392,6 +1392,7 @@ interface(`kernel_read_core_if',`
 	')
 
 	allow $1 self:capability sys_rawio;
+	allow $1 self:lockdown confidentiality;
```
Could you split this out into a separate commit? (It doesn't seem directly related to the unsigned module loading.)

zpytela committed:
Add confidentiality lockdown permission to kernel_read_core_if()

Since adding the lockdown class to selinux-policy, the confidentiality
permission starts to be checked on each access of /proc/kcore.
The permission to read the core kernel interface is allowed in the
kernel_read_core_if() interface, so the confidentiality lockdown
permission was also added to this interface.

zpytela commented:
Commits split, commit message updated.

WOnder93 commented on policy/modules/contrib/kdump.te:
```diff
@@ -63,6 +63,7 @@ files_read_kernel_modules(kdump_t)
 files_read_kernel_img(kdump_t)
 files_map_boot_files(kdump_t)
 
+kernel_load_unsigned_module(kdump_t)
```
I looked up the RHEL-9 BZ behind this and the operation that kdump is doing is not a kernel module load, but rather `kexec_load(2)`, which allows to specify a kernel image to reboot into (usually this is used to do analysis/memory dump after a system crash). This currently doesn't support signature verification of the kernel being loaded, so it always asks for lockdown integrity permission.

I'd prefer to create a new interface for this (e.g. `kernel_kexec_load()`), which would grant both the `sys_boot` capability ([also required unconditionally for this syscall](https://elixir.bootlin.com/linux/v5.11/source/kernel/kexec.c#L204)) and the `lockdown` `integrity` permission.

WOnder93 commented on policy/modules/kernel/kernel.if:
```diff
@@ -4283,3 +4284,23 @@ interface(`kernel_unlabeled_entry_type',`
+## </param>
+#
+interface(`kernel_kexec_load',`
+	gen_require(`
+		type unlabeled_t;
+	')
```
Forgot to remove this?

zpytela replied:
A result of rebasing when the block appeared back.

WOnder93 commented on policy/modules/kernel/kernel.if:
```diff
@@ -4283,3 +4284,19 @@ interface(`kernel_unlabeled_entry_type',`
 	allow $1 unlabeled_t:file { mmap_exec_file_perms ioctl lock };
 ')
 
+
```
This is just a cosmetic detail, but there is an extra empty line being added here. And there seems to be also an extra empty line being added to the end of file.

zpytela replied:
Updated.

WOnder93 reviewed (commented):
Thanks, looks good now!

WOnder93 reviewed (approved):
Sorry, meant to approve, not comment...
